saq/queue.py

```diff
@@ -403,3 +439,127 @@ async def enqueue(self, job_or_func, **kwargs):
 
         logger.info("Enqueuing %s", job)
         return job
+
+    async def apply(self, job_or_func, timeout=None, **kwargs):
+        """
+        Enqueue a job and wait for its result.
+
+        If the job is successful, this returns its result.
+        If the job is unsuccessful, this raises a JobError.
+
+        Example::
+            try:
+                assert await queue.apply("add", a=1, b=2) == 3
+            except JobError:
+                print("job failed")
+
+        job_or_func: Same as Queue.enqueue
+        kwargs: Same as Queue.enqueue
+        """
+        results = await self.map(job_or_func, timeout=timeout, iter_kwargs=[kwargs])
+        return results[0]
+
+    async def map(self, job_or_func, iter_kwargs, timeout=None, **kwargs):
+        """
+        Enqueue multiple jobs and collect all of their results.
+
+        This raises a JobError if any of the jobs are unsuccessful.
+
+        This waits for job results in the order they are enqueued, so JobExceptions are
+        not necessarily raised immediately.
+
+        Example::
+            try:
+                assert await queue.map(
+                    "add",
+                    [
+                        {"a": 1, "b": 2},
+                        {"a": 3, "b": 4},
+                    ]
+                ) == [3, 7]
+            except JobError:
+                print("any of the jobs failed")
+
+        job_or_func: Same as Queue.enqueue
+        iter_kwargs: Enqueue a job for each item in this sequence. Each item is the same
+            as kwargs for Queue.enqueue.
+        timeouts: Total seconds to wait for all jobs to complete. If None or 0, wait forever.
+        kwargs: Default kwargs for all jobs. These will be overridden by those in iter_kwargs.
+        """
+        jobs = [self.prepare_job(job_or_func, **{"timeout": timeout, **kwargs, **kw}) for kw in iter_kwargs]
+        jobs_by_id = {job.id: job for job in jobs}
+
+        done = 0
+        failed_job = None
+
+        async def callback(_id, status):
```

Comment:
change to job_id

Reply:
Ah, sure, that's better.
I copy&pasted this from https://github.com/tobymao/saq/blob/master/saq/job.py#L205